src/magic_folder/cli.py

```diff
@@ -553,11 +557,13 @@ def config(self):
 
     @property
     def client(self):
-        if self._http_client is None:
-            from twisted.internet import reactor
-            self._http_client = create_http_client(reactor, self.config.api_client_endpoint)
         if self._client is None:
             from twisted.internet import reactor
+
+            if self._http_client is None:
+                self._http_client = create_http_client(
+                    reactor, self.config.api_client_endpoint
+                )
```

Comment:
Since this is only used if we don't already have a client, there is no need to create one if we do have a client.

I had made this change, because I thought I had seen a test in `magic_folder.test.test_api_cli` that set it, but it appears I was mistaken. (Those tests newly hit this class, since I changed the options class there to inherit from this one.